### DISCUSSION TITLE
Error: Can't resolve 'remark' in '/vercel/workpath0/lib'

### DISCUSSION THREAD

**Discussion by bomeers** [vercel/next.js#18176] [Help]
Seems I am getting a compile error as shown in the title.

Here is my repo: https://github.com/bomeers/nextjs-blog

Here is the error code:
```

Failed to compile.
--
15:39:10.108 | ModuleNotFoundError: Module not found: Error: Can't resolve 'remark' in '/vercel/workpath0/lib'
15:39:10.108 | > Build error occurred
15:39:10.109 | Error: > Build failed because of webpack errors
15:39:10.109 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:15:918)
15:39:10.109 | at runMicrotasks (<anonymous>)
15:39:10.109 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
15:39:10.121 | npm ERR! code ELIFECYCLE
15:39:10.121 | npm ERR! errno 1
15:39:10.122 | npm ERR! learn-starter@0.1.0 build: `next build`
15:39:10.122 | npm ERR! Exit status 1
15:39:10.122 | npm ERR!
15:39:10.123 | npm ERR! Failed at the learn-starter@0.1.0 build script.
15:39:10.123 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
15:39:10.128 | npm ERR! A complete log of this run can be found in:
15:39:10.128 | npm ERR!     /vercel/.npm/_logs/2020-10-23T20_39_10_123Z-debug.log
15:39:10.132 | Error: Command "npm run build" exited with 1

```

I've tried using [this method](https://github.com/vercel/next.js/discussions/14416#discussioncomment-28893) and it worked for a previous similar build issue with the `date-fns` package, ~~but it will not work with this one.~~

**Update:** Seems that it needed time to re-deploy from github. It is working properly now with the method stated above.

**Comment by SoorajChandran** [comment]
I'm getting the same error. But the file missing is a mu component that I built. Doesn't fail when I'm building locally thought. Not sure what is wrong?

```

Failed to compile.
--
19:28:59.824 | ModuleNotFoundError: Module not found: Error: Can't resolve '../components/navigation' in '/vercel/workpath0/pages'
19:28:59.825 | > Build error occurred
19:28:59.826 | Error: > Build failed because of webpack errors
19:28:59.826 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:15:918)
19:28:59.826 | at runMicrotasks (<anonymous>)
19:28:59.826 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
19:28:59.842 | error Command failed with exit code 1.

```

**Comment by toke182** [comment]
I am having the exact same issue, I have my component working, the import is correct and it builds in my local perfectly.

It just fails in vercel CI, someone managed to find out why?
```
23:54:47.582 | Failed to compile.
23:54:47.582 | ModuleNotFoundError: Module not found: Error: Can't resolve '../LikeButton/likebutton' in '/vercel/workpath0/components/PostCard'
23:54:47.582 | > Build error occurred
23:54:47.583 | Error: > Build failed because of webpack errors
23:54:47.583 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:15:918)
23:54:47.583 | at runMicrotasks (<anonymous>)
23:54:47.583 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
23:54:47.603 | error Command failed with exit code 1.
23:54:47.606 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
23:54:47.613 | Error: Command "yarn run build" exited with 1
```

**Reply by StarDrop9** [reply]
Ditto

18:51:02.097 | Failed to compile.
-- | --
18:51:02.097 | ModuleNotFoundError: Module not found: Error: Can't resolve '../components/layout' in '/vercel/workpath0/pages'
18:51:02.098 | > Build error occurred
18:51:02.099 | Error: > Build failed because of webpack errors
18:51:02.099 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:15:918)
18:51:02.099 | at runMicrotasks (<anonymous>)
18:51:02.099 | at processTicksAndRejections (internal/process/task_queues.js:97:5)

**Comment by dbudimir** [comment]
Same issue here... No problems running vercel --prod, but I get this error when I try to deploy through GitHub.

20:58:28.118 | Failed to compile.
20:58:28.119 | ModuleNotFoundError: Module not found: Error: Can't resolve '../components/Nav' in '/vercel/workpath0/pages'
20:58:28.119 | > Build error occurred
20:58:28.120 | Error: > Build failed because of webpack errors
20:58:28.120 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:15:918)
20:58:28.120 | at runMicrotasks (<anonymous>)
20:58:28.120 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
20:58:28.141 | error Command failed with exit code 1.
20:58:28.141 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
20:58:28.154 | Error: Command "yarn run build" exited with 1
20:58:32.191 | Done with "package.json"

**Reply by StarDrop9** [reply]
Oh right. In dev my .js component files did not have the extension on them. Ran fine. Deployed to Vercel apparently I needed to add the extension ".js" to each component file. Probably that might be an issue or compatible "Case" also. Upper or lower may be off in the code. Pretty pickie on Vercel.

**Reply by FelipeMaeda** [reply]
I had the same issue, but im my case i verified that Upper Letter can prevent the deploy. Bellow my code of import:

Before:
import Layout from '../components/Layout/menu'

After:
import Layout from '../components/layout/menu'

Att,
Felipe Maeda

**Reply by acosta-dev** [reply]
"did not have the extension..." save the day, that's where my problem was, thanks.!

**Reply by urielbitton** [reply]
I have same errors but it can't be because of the .js extensions that are missing because my entire project ignores extensions and i'm only getting the error when i uncomment a specific set of imports...

**Comment by JeredOdegard** [comment]
Same exact issue. It works locally, in dev and with `vercel --prod` but fails to deploy to Vercel from Github. I receive `ModuleNotFoundError: Module not found: Error: Can't resolve 'components/navbar'` when trying to deploy. After a few hours of trial and error, I was able to get it working -- somehow. Not sure why this worked. It sounds like OP's was fixed after some time? I changed:

`import NavBar from "components/navbar"` to `import NavBar from "components/NavBar.js"`

_(matching the camel-case and adding the .js extension)_

**\*\*\*EDIT\*\*\***

It turns out, the error occurred for me when I changed the casing on a few file names e.g.,`/File.js` to `/file.js`. Because `git config caseignore` was `true`, it did not propagate to Github, then when I did change the config, it made duplicates on the remote. Don't do that! The best way to fix the original issue, my duplication, and when you're not sure which files could be contaminated is to remove the git tree, re-stage, and push to Github. See example below:

`git rm -r --cached .`
`git add --all .`
`git commit -am "Fix casing discrepancies."`
`git push origin branch_name`

Alternatively, if you changed casing on a single file (or just a few), you can run `git mv` on each one like this...

`git mv -f components/oldfile.js components/OldFile.js`
`git mv -f components/OldFile.js components/oldfile.js`

...and then stage, commit, and push to Github. If you did create duplicates like me, you can check `git status` after running the `git mv` commands above. You should see a deletion statement of the duplicate file.

**Reply by dbudimir** [reply]
Legend! Thank you for this, solved it for me as well.

**Reply by petehanssens** [reply]
this is awesome - thanks so much!

**Reply by truejoolean** [reply]
life saver! Thanks

**Reply by S3gillu** [reply]
This is what I was looking for. Awesome explanation

**Reply by Pkfication** [reply]
Nice!!! Worked for me as well!

**Reply by Kais3rP** [reply]
Yeah, this fixed it!

**Reply by SSylvain1989** [reply]
Works fine i change all my import from `../components/Layout` to `../components/Layout.js ` Thanks

**Reply by laphilosophia** [reply]
thanks a lot! this fixed it

**Reply by paulonteri** [reply]
Thanks man!

**Reply by xodeeq** [reply]
This worked for me mate, big thanks.

**Reply by jonatasvnascimento** [reply]
Saved me, the component name was without .js and in lowercase letters.
On a local server it didn't differentiate (it gave an error), but on vercel it kept giving this error

**Reply by jonathanlurie** [reply]
This saved me! Thanks a lot!

**Reply by passandscore** [reply]
Life Saver! Much appreciated.

**Reply by ouweiya** [reply]
Everything is ready, ready to launch, but I tripped over the file name error.

![aaa](https://user-images.githubusercontent.com/28357731/144044530-b028b9a0-a8ad-46d2-9e0e-0347f3008f5c.gif)

**Reply by AminAbdisamad** [reply]
Thank you @JeredOdegard. This helped me a lot.

**Reply by Pra-Shant-Chouhan** [reply]
Thanks lot man You saved my  carrier

**Reply by renoeno** [reply]
thanks a lot! this was exactly what was happening with me.

**Reply by tms1337** [reply]
never would have think of this, thanks!

**Reply by KhaledDerkaoui** [reply]
Thanks a lot bro

**Reply by vladimirplyukhin89** [reply]
Thanks a lot!

**Reply by bravo680git** [reply]
thank you, it is very useful

**Reply by haracena** [reply]
God bless you! Thank you so much friend.

**Reply by MaxHuynh-Dev** [reply]
Thanks for your way help me a lot

**Reply by MadsFrost** [reply]
Thank you so much aswell :)

**Reply by halitak** [reply]
thanks

**Reply by BaggioGiacomo** [reply]
Thank you, very useful! 🚀

**Reply by walterfcarvalho** [reply]
This works fine for me. 

> Same exact issue. It works locally, in dev and with `vercel --prod` but fails to deploy to Vercel from Github. I receive `ModuleNotFoundError: Module not found: Error: Can't resolve 'components/navbar'` when trying to deploy. After a few hours of trial and error, I was able to get it working -- somehow. Not sure why this worked. It sounds like OP's was fixed after some time? I changed:
> 
> `import NavBar from "components/navbar"` to `import NavBar from "components/NavBar.js"`
> 
> _(matching the camel-case and adding the .js extension)_
> 
> *****EDIT*****
> 
> It turns out, the error occurred for me when I changed the casing on a few file names e.g.,`/File.js` to `/file.js`. Because `git config caseignore` was `true`, it did not propagate to Github, then when I did change the config, it made duplicates on the remote. Don't do that! The best way to fix the original issue, my duplication, and when you're not sure which files could be contaminated is to remove the git tree, re-stage, and push to Github. See example below:
> 
> `git rm -r --cached .` `git add --all .` `git commit -am "Fix casing discrepancies."` `git push origin branch_name`
> 
> Alternatively, if you changed casing on a single file (or just a few), you can run `git mv` on each one like this...
> 
> `git mv -f components/oldfile.js components/OldFile.js` `git mv -f components/OldFile.js components/oldfile.js`
> 
> ...and then stage, commit, and push to Github. If you did create duplicates like me, you can check `git status` after running the `git mv` commands above. You should see a deletion statement of the duplicate file.

**Reply by CinArb2** [reply]
thank you so much. It worked for me too. 👍

**Reply by content-turbine-editor** [reply]
Thank you so muchh. Works like a charm.

**Reply by daviemola** [reply]
Thanks a lot. I had similar issues trying to deploy on both vercel and Netlify. Much appreciated.

**Reply by noam-elbaz** [reply]
Thanks! worked perfectly.

**Reply by toniengelhardt** [reply]
Wow, what a mad error, thanks so much 🙏🏽 🙏🏽 🙏🏽

**Reply by Rezet2000** [reply]
Thank you so much! worked for me.

**Reply by ayhanap** [reply]
Thanks, luckily it was just a couple of files so I used `git mv -f`. But I would say this article is very misleading at all https://vercel.com/guides/how-do-i-resolve-a-module-not-found-error.

**Reply by Cristian327** [reply]
you are the best

**Comment by danielsimao** [comment]
Hi guys, I had the same problem. I had a link in my .env for iframe src that was breaking the build, not really sure why.

### MY SUGGESTION is to search in the code for the keyword mentioned in the log and start from there. 

For my problem it was "n18m", I removed the link from the .env and the build went green. Cheers

**Reply by promiseoomos** [reply]
Thank you! Thank you! Thank you! works very well

**Comment by Deep-Codes** [comment]
The problem with most of `not resolve <Module Name>` is the casing I had build fail with `cannot resolve Prismjs` the problem here was I used `import Prism from "Prismjs";` instead of `import Prism from "prismjs";`

<img width="536" alt="Screenshot 2020-12-25 at 4 33 53 PM" src="https://user-images.githubusercontent.com/61158210/103133197-09b58e00-46cf-11eb-9340-16aab13ab332.png">

Made these Changes and the Vercel Deploy was Succesful
 
Builds don't face issue locally as mine is MacOs but Vercel/Netlify use Linux for Builds which are Case Sensitive so maybe that be a reason. Hope that Helps!🎉

**Reply by yuxxeun** [reply]
thanks a lot, its just work on my machine.

**Comment by GaddMaster** [comment]
I had same issue , I changed some them component names back and forth and all good

**Comment by withoutwax** [comment]
For some reason, the solutions outlined here didn't work for me - but fortunately, implementing Absolute Imports and Module path aliases fixed my issue! Just wanted to share it here for anyone facing similar problem!

For implementing this, visit the Next.js documentation:
[https://nextjs.org/docs/advanced-features/module-path-aliases](https://nextjs.org/docs/advanced-features/module-path-aliases)

**Comment by raihan-ibiraeva** [comment]
Compare the folder name in VS code and GitHub, in my case it was different. Renamed the folders to match(it's a 2 step process if you want to keep the same word - temp rename, then rename back with correct casing) and that fixed the issue.

**Reply by muttenzer** [reply]
✔️  Thank you so much! This drove me almost crazy…

**Reply by sloanlacey** [reply]
Thank you! I had one lowercase letter out of place and this helped me find it.

**Comment by danroberts** [comment]
It seems wrong to me that the local development server allows case insensitive imports but the production build does not. Even running `npm run build` produces no errors, but then fails on the deploy. I would actually expect the development server to throw an error if I try to import something with the wrong case sensitivity.

**Comment by mattburnett-repo** [comment]
Hi

A little late to the party here, but after dealing with this exact problem I thought I'd share another possible solution:

In my case, there was no entry for either "remark" or "remark-html" in the package.json file in the project root. 

I added

"remark": "13.0.0",
"remark-html": "13.0.1"

to the "dependencies" section and re-ran the build / deploy successfully.

Hope this helps :)

**Comment by sagar2agrawal** [comment]
The issue is very simple,

In **windows**, the file name 'component' and 'Component' are considered as same, so its **case insensitive**,

But the Vercel server might be using Linux-based OS, which is a case-sensitive means 'component' name and 'Component' name file are considered different.

Hence in any import, even if the name is the same for any of the files or folders but the case is different, then it will give the error

**Comment by BleddP** [comment]
I had the same issue where it would fail on some components, but not on others. After changing everything to lowercase where needed, it still didn't fix the issue. 

Eventually I changed my file names from .jsx to .js and that seemed to do the trick. Not sure why but it works for me

**Reply by tlanettepollard** [reply]
I had the same problem and that fixed it for me. I had been sitting here for an hour until I found this thread. I changed mine from .jsx to .js.

**Comment by anupam-io** [comment]
For me, `npm run build` and `npm run start` are working fine in my machine. 
But vercel deployment is failing:

```
Failed to compile.
--
23:41:13.067 |  
23:41:13.068 | ModuleNotFoundError: Module not found: Error: Can't resolve '/vercel/path1/node_modules/next/node_modules/@babel/runtime/helpers/construct' in '/vercel/path1/node_modules/next/dist/client'
23:41:13.068 |  
23:41:13.068 |  
23:41:13.068 | > Build error occurred
23:41:13.069 | Error: > Build failed because of webpack errors
23:41:13.069 | at /vercel/path1/node_modules/next/dist/build/index.js:397:19
23:41:13.069 | at runMicrotasks (<anonymous>)
23:41:13.070 | at processTicksAndRejections (internal/process/task_queues.js:95:5)
23:41:13.070 | at async Span.traceAsyncFn (/vercel/path1/node_modules/next/dist/telemetry/trace/trace.js:60:20)
23:41:13.070 | at async Object.build [as default] (/vercel/path1/node_modules/next/dist/build/index.js:77:25)

```

**Reply by Pra-Shant-Chouhan** [reply]
use full path in your github import like
ex- import Header from './Layout/coponentHedFoot/Header.js'
extension of every filepath add

and remamber first Letter of file name should be capital
